### DISCUSSION TITLE
scoping configuration key of modules

### DISCUSSION THREAD

**Discussion by geoHeil** [nuxt/nuxt#20584] [Questions]
How can I scope the configuration of nuxt plugins / modules to be inside the runtimeConfig group?

https://github.com/xlanex6/nuxt-meilisearch/issues/26

**Comment by maou-shonen** [comment]
I don't use this module.
but it seems to be included [here](https://github.com/xlanex6/nuxt-meilisearch/blob/0aaa22ce5fa1569d2e36960eaf3a4a329fe920fc/src/module.ts#LL64C59-L64C82).
You should be able to override it with env `NUXT_MEILISEARCH_CLIENT_XXXX`.

**Reply by geoHeil** [reply]
What is the XXX? and how would a move of the `meilisearch` inside the `runtimeConfig` look like? Trying to set a different value like xxx (still top-level and not yet moved inside) via NUXT_MEILISEARCH_CLIEN fails.

How should this work?

**Reply by maou-shonen** [reply]
In your example this should be `NUXT_SERVER_MEILISEARCH_CLIENT_ADMIN_API_KEY`,
If that doesn't work, you can create a plugin instead to override.
https://nuxt.com/docs/guide/going-further/runtime-config#environment-variables

**Reply by geoHeil** [reply]
Understood. I thought I have to move the scope. If this works - then this would be a solution. However, this is not working for me.

Inside `nuxt.config.ts`: 

```
meilisearch: {
    hostUrl: "https://search.foo.bar",
```

is set to a dummy value.

Now when providing:

```
NUXT_SERVER_MEILISEARCH_CLIENT_HOST_URL=https://search.real-domain.com 
```

nuxt is still using the old dummy value. it is not overwriting it.

**Reply by maou-shonen** [reply]
Have you tried create a plugin to override it?
it should work

**Reply by geoHeil** [reply]
Not yet, as it is still unclear to me how to register the hook (instantiation of the plugin) and configuration with my values https://nuxt.com/docs/guide/going-further/runtime-config#plugins

**Reply by xlanex6** [reply]
it could be 

```ts
meilisearch: {
    hostUrl:  process.env.NUXT_SERVER_MEILISEARCH_CLIENT_HOST_URL,
... 
```

env file
```
NUXT_SERVER_MEILISEARCH_CLIENT_HOST_URL=https.......
```

and set runtime config like so
NUXT_SERVER_MEILISEARCH_CLIENT_HOST_URL shoulb be in public  like describe here
more info on https://nuxt.com/docs/guide/going-further/runtime-config#exposing-runtime-config

**Reply by maou-shonen** [reply]
Yes, you can try again using environment variables with the `NUXT_` prefix.
In Nuxt 3.5 (Nitro 2.4), useRuntimeConfig is parse on each request,
but you still need to set a default value to pass the non-empty check of the module.

**Reply by geoHeil** [reply]
> it could be
> 
> ```ts
> meilisearch: {
>     hostUrl:  process.env.NUXT_SERVER_MEILISEARCH_CLIENT_HOST_URL,
> ... 
> ```
> 
> env file
> 
> ```
> NUXT_SERVER_MEILISEARCH_CLIENT_HOST_URL=https.......
> ```
> 
> and set runtime config like so NUXT_SERVER_MEILISEARCH_CLIENT_HOST_URL shoulb be in public like describe here more info on https://nuxt.com/docs/guide/going-further/runtime-config#exposing-runtime-config

So you mean it should be referenced twice? once in the global meilisearch block? and a 2nd time inside the public block?
